SIGSEGV on `ttn-lw-cli device delete` 3.20.0

<!--
Thanks for submitting an issue. Please fill the template below,
otherwise we will not be able to process this issue.
-->

#### Summary
<!-- Summarize the issue in a few sentences: -->
Deleting an end-device using `ttn-lw-cli` can throw a SIGSEGV.

#### Steps to Reproduce
<!-- How can we reproduce the problem? How should the feature work? -->
Not 100% sure what the trigger is (have deleted a device with this code before).

What I did in this case:
1. create the test device using `ttn-lw-cli`
2. deleted the test device in the console
3. created the test device in the console (same name, but in ABP mode)
4. attempt to run a script that identified ABP mode device, deletes it before recreating as OTAA.

The command run was:

``` bash
ttn-lw-cli --config "${ipath}/ttnconfig/ttn-lw-cli-lcs.yml" device delete --application-id ${app_id} --device-id ${dev_id}
```
_NB: I just changed absolute values to bash syntax variables to abstract personal details out of the invocation. The values were valid and correct.  I've used them on the CLI in other contexts to create this device before._

The response was:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x11acb74]

goroutine 1 [running]:
go.thethings.network/lorawan-stack/v3/cmd/ttn-lw-cli/commands.glob..func85(0x2544958, {0x2dbc3c0, 0x0, 0x6})
  /home/runner/work/lorawan-stack/lorawan-stack/cmd/ttn-lw-cli/commands/end_devices.go:1014 +0x498
github.com/spf13/cobra.(*Command).execute(0x2544958, {0x2dbc390, 0x6, 0x6})
  /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.2.1/command.go:856 +0x6b0
github.com/spf13/cobra.(*Command).ExecuteC(0x254b4f8)
  /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.2.1/command.go:974 +0x414
main.main()
  /home/runner/work/lorawan-stack/lorawan-stack/cmd/ttn-lw-cli/main.go:27 +0x20
```


#### Environment
<!--
Your environment: OS/Browser/Gateway/Device/...? Versions? IDs/EUIs?
Paste the output of "ttn-lw-cli version" or "ttn-lw-stack version" if applicable.
-->
This was using the TTN community server: the eu1.cloud.thethings.network:8884

Test host is running 'Raspbian':
```
The Things Network Command-line Interface: ttn-lw-cli
Version:             3.20.0
Build date:          2022-06-17T09:38:56Z
Git commit:          899adb340
Go version:          go1.18.3
OS/Arch:             linux/arm
```

#### How do you propose to implement this?
Someone familiar with the code looking at the stack trace and working out how a nil pointer is being abused with no error checking or appropriate try/catch.

Should  further logs or access to the test device be useful I'll work with an interested party to get access.

Sorry not to fix it myself.  Hope this report is still a helpful clue!

Thanks for the hard work!

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -31,6 +31,7 @@ For details about compatibility between different releases, see the **Commitment
 ### Fixed
 
 - Console showing `404 Not Found` errors for pages containing user IDs in the path, when the user ID has a length of two.
+- CLI no longer panics when deleting a device without JoinEUI, this scenario only occurred when deleting a device that uses ABP.
 
 ### Security
 

--- a/cmd/ttn-lw-cli/commands/end_devices.go
+++ b/cmd/ttn-lw-cli/commands/end_devices.go
@@ -1004,7 +1004,7 @@ var (
 				return errAddressMismatchEndDevice.New()
 			}
 
-			if existingDevice.JoinServerAddress == "" {
+			if existingDevice.JoinServerAddress == "" && devID.GetJoinEui() != nil {
 				// Attempt to unclaim device via the DCS.
 				dcs, err := api.Dial(ctx, config.DeviceClaimingServerGRPCAddress)
 				if err != nil {

